display/d.text/test.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+# Rewritten from test.pl
+
+import math
+import re
+
+# Quiet black syntax checking for fonts and colors to keep each respective tuple
+# in the following format when rendering.
+# fmt: off
+fonts = ("cyrilc", "gothgbt", "gothgrt", "gothitt", "greekc", "greekcs",
+         "greekp", "greeks", "italicc", "italiccs", "italict", "romanc",
+         "romancs", "romand", "romans", "romant", "scriptc", "scripts",
+         "cyrilc", "gothgbtlu")
+colors = ("red", "orange", "yellow", "green", "blue", "indigo", "violet",
+          "black", "gray", "brown", "magenta", "aqua", "grey", "cyan",
+          "purple")
+# fmt: on
+
+
+def rc(r, c):
+    print(f".X {r}\n.Y {c}")
+
+
+def xy(x, y):
+    print(f".X {x}%\n.Y {y}%")
+
+
+def font(f):
+    print(f".F {f}")
+
+
+def size(s):
+    print(f".S {s}")
+
+
+def color(c):
+    print(f".C {c}")
+
+
+def rotate(r):
+    print(f".R {r}")
+
+
+def align(a):
+    print(f".A {a}")
+
+
+def text(in_text):
+    print(f"{in_text}")
+
+
+for i in range(36):
+    font(fonts[int(i % len(fonts))])
+    size(((36 - i if (i >= 9 and i <= 18 or i > 27) else i) % 9))
+    rotate(i * 10)
+    color(colors[i % len(colors)])
+    xy(
+        (80 + 10 * math.cos(i * 10 / 180 * 3.141593)),
+        (50 + 10 * 640 / 480 * math.sin(i * 10 / 180 * 3.141593)),
+    )
+    text(fonts[int(i % len(fonts))])
+
+size(2)
+rotate(0)
+font("romans")
+color("gray")
+rc(1, 1)
+
+with open(__file__) as f:
+    src = f.read()
+    "\n.L 1\n.L 0\n".join(f.read().split("\n")[1:])
+
+print(
+    ".L 0\n"
+    + re.sub(
+        '(".*?")',
```

Comment:
This is actually interesting. Black complained about single-quote indexing (#1425), but it leaves regular single-quote strings. It even reformats escaped double quotes inside double quotes. Another confusing lesson from Black.

Reply:
Yes, I was going to leave a comment pointing this out  since you addressed the want for double quotations. Glad you noticed it.

Reply:
Ad quoting, Black abandons(x) the practice of single quotes for indexes/values and double quotes for user strings/output. Instead, it uses the classic/original reasoning for single and double quotes - to limit the need for escaping when quotes are in the strings when, e.g., mixing languages. This has more clear rules compared to the other practice. The point with using Black is: you just do something reasonable and let Black do its thing automatically, then you check the checks in a PR that you didn't forget to run Black. No figuring out what is the proper/project/file/Black style. When everything works as it should (in terms of Black versions, for example), it is not you trying to get the Black formatting right, it is Black taking care of the formatting for you.

x Actually, it is the default settings we decided to use. It has legacy option for project which don't want to switch.